include/groonga/groonga.h

```diff
@@ -1827,9 +1844,10 @@ grn_ctx_recv_handler_set(grn_ctx *,
 /* various values exchanged via grn_obj */
 
 #define GRN_OBJ_DO_SHALLOW_COPY (GRN_OBJ_REFER | GRN_OBJ_OUTPLACE)
-#define GRN_OBJ_VECTOR          (0x01 << 7)
+/// A flag indicating a vector type object.
```

Comment:
```suggestion
/// A flag to initialize a vector object.
```

実は、`GRN_OBJ_VECTOR`ってベクターのオブジェクトについているフラグじゃないんだよね。
`GRN_XXX_INIT()`とか`grn_obj_reinit()`とかで初期化するときにベクターにしたいというのを示すときと、`grn_obj_get_range_info()`でベクターカラムかどうかを返すときだけに使っているんだよ。

前者のときは最終的には`GRN_PVECTOR`とか`GRN_VECTOR`とかという別の情報が使われて`GRN_OBJ_VECTOR`というフラグは残らなくて、
後者のときは`GRN_OBJ_COLUMN_VECTOR`フラグがあったら`GRN_OBJ_VECTOR`として返しているだけなんだよ。

後者のときに`GRN_OBJ_VECTOR`として返しているのは`grn_obj_reinit()`とかで使いやすいようにしたいから。

Reply:
fix: [4a894ba](https://github.com/groonga/groonga/pull/1944/commits/4a894bab2ec784169a154391311b8ff41d36190a) なるほどです！
`GRN_OBJ_VECTOR` はベクターしてねという支持をするフラグなのであって、このオブジェクトがベクターであると表現するものではないと理解しました。